lib/model/indexhandler.go

```diff
@@ -328,12 +348,42 @@ func (s *indexHandler) receive(fs []protocol.FileInfo, update bool, op string) e
 	if !update {
 		fset.Drop(deviceID)
 	}
+
+	l.Debugf("Received %d files for %s from %s, prevSeq=%d, lastSeq=%d", len(fs), s.folder, deviceID.Short(), prevSequence, lastSequence)
+
+	// Verify that the previous sequence number matches what we expected
+	if prevSequence > 0 && prevSequence != fset.Sequence(deviceID) {
+		l.Warnf("Bug: device %v folder %s sent index update with sequence %d while we expected %d", deviceID.Short(), s.folder, fset.Sequence(deviceID), s.prevSequence)
```

Comment:
All these warnings will become banners in the UI, and there's potential for a lot of them - both if this uncovers a (the! :) ) bug or if we get something about the debugging wrong. I assume the intention is for the user to bring such reports to the forum, so we get someone to debug with in addition to just the reports, that we already get from failure reporting. In that case lets maybe do just a single warning per failure type and device on first occurrence, and then info log to avoid plastering the GUI?